lib/msal-common/src/client/SilentFlowClient.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { BaseClient } from "./BaseClient";
+import { ClientConfiguration } from "../config/ClientConfiguration";
+import { SilentFlowRequest } from "../request/SilentFlowRequest";
+import { AuthenticationResult } from "../response/AuthenticationResult";
+import { CredentialType } from "../utils/Constants";
+import { CacheRecord } from "../unifiedCache/entities/CacheRecord";
+import { IdTokenEntity } from "../unifiedCache/entities/IdTokenEntity";
+import { CacheHelper } from "../unifiedCache/utils/CacheHelper";
+import { AccessTokenEntity } from "../unifiedCache/entities/AccessTokenEntity";
+import { RefreshTokenEntity } from "../unifiedCache/entities/RefreshTokenEntity";
+import { ScopeSet } from "../request/ScopeSet";
+import { IdToken } from "../account/IdToken";
+import { TimeUtils } from "../utils/TimeUtils";
+import { RefreshTokenRequest } from "../request/RefreshTokenRequest";
+import { RefreshTokenClient } from "./RefreshTokenClient";
+import { ClientAuthError } from "../error/ClientAuthError";
+import { CredentialFilter, CredentialCache } from "../unifiedCache/utils/CacheTypes";
+
+export class SilentFlowClient extends BaseClient {
+
+    constructor(configuration: ClientConfiguration) {
+        super(configuration);
+    }
+
+    /**
+     * Retrieves a token from cache if it is still valid, or uses the cached refresh token to renew
+     * the given token and returns the renewed token
+     * @param request
+     */
+    public async acquireToken(request: SilentFlowRequest): Promise<AuthenticationResult>{
+        let cacheRecord: CacheRecord;
+        let idTokenObj: IdToken;
+        const requestScopes = new ScopeSet(request.scopes || [], this.config.authOptions.clientId, true);
+
+        // We currently do not support silent flow for account === null use cases; This will be revisited for confidential flow usecases
+        if (request.account === null) {
```

Comment:
Can this be `if (!request.account)`? Do you need to explicitly check for `null`?

Reply:
done.